cpp/src/arrow/c/bridge.cc

```diff
@@ -1868,24 +1868,15 @@ struct ArrayImporter {
   template <typename OffsetType>
   Status ImportStringValuesBuffer(int32_t offsets_buffer_id, int32_t buffer_id,
                                   int64_t byte_width = 1) {
-    if (device_type_ == DeviceAllocationType::kCPU) {
-      auto offsets = data_->GetValues<OffsetType>(offsets_buffer_id);
-      // Compute visible size of buffer
-      int64_t buffer_size =
-          (c_struct_->length > 0) ? byte_width * offsets[c_struct_->length] : 0;
-      return ImportBuffer(buffer_id, buffer_size);
-    }
-
-    // we only need the value of the last offset so let's just copy that
-    // one value from device to host.
-    auto single_value_buf =
-        SliceBuffer(data_->buffers[offsets_buffer_id],
-                    c_struct_->length * sizeof(OffsetType), sizeof(OffsetType));
-    ARROW_ASSIGN_OR_RAISE(
-        auto cpubuf, Buffer::ViewOrCopy(single_value_buf, default_cpu_memory_manager()));
-    auto offsets = cpubuf->data_as<OffsetType>();
+    int64_t last_offset_value_offset =
+        c_struct_->length * sizeof(OffsetType) + c_struct_->offset;
```

Comment:
@felipecrv @zeroshade just want to make sure I'm meant to add the ```c_struct_->offset``` here?

I think the last offset value will be data() + buffer offset (c_struct_->offset()) + item offset but thats different from the ```SliceBuffer``` call that was previously here

Reply:
Yes. The offset of the ArrowArray should be applied when reading the `Buffer` of the array.

But the formula is different because the array offset is in number of values, not in the number of bytes.

`(c_struct_->offset + c_struct_->length) * sizeof(OffsetType)`

Reply:
Ah yep that makes sense